server/sonar-server/src/main/java/org/sonar/server/db/BaseDao.java

```diff
@@ -215,7 +215,7 @@ private void update(DbSession session, DTO item, Date now) {
       item.setUpdatedAt(now);
       doUpdate(session, item);
       if (hasIndex()) {
-        session.enqueue(new UpsertDto(getIndexType(), item));
+        session.enqueue(new UpsertDto<DTO>(getIndexType(), item));
```

Comment:
For your information BaseDao is being dropped. No need to make effort to improve it.